src/framework/chart/LineChart.jsx

```diff
@@ -163,6 +168,66 @@ export default class LineChart extends Component {
     // Transition each line from the previous shape to the new shape.
     lines.transition().duration(duration)
       .attr('d', item => lineGenerator(item.values));
+
+    // TOOLTIP /////////////////////////////////////////////////////////////////
```

Comment:
In the context of the renderChart method it doesn't make sense to only extract that tooltip part but we might want to think of extracting parts of the method in general.